drivers/soundwire/bus_type.c

```diff
@@ -174,9 +178,15 @@ static void sdw_drv_shutdown(struct device *dev)
  */
 int __sdw_register_driver(struct sdw_driver *drv, struct module *owner)
 {
+	const char *name;
+
 	drv->driver.bus = &sdw_bus_type;
 
 	if (!drv->probe) {
+		name = drv->name;
+		if (!name)
+			name = drv->driver.name;
+
 		pr_err("driver %s didn't provide SDW probe routine\n",
 		       drv->name);
```

Comment:
didn't you mean to replace `drv->name` here too?

Reply:
yep, good catch @lyakh !

Reply:
fixed now